include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -119,8 +124,8 @@ class cpp_dec_float
  private:
    using array_type =
       typename std::conditional<std::is_void<Allocator>::value,
-                                detail::static_array <std::uint32_t, cpp_dec_float_elem_number>,
-                                detail::dynamic_array<std::uint32_t, cpp_dec_float_elem_number, Allocator> >::type;
+                                detail::static_array <std::uint32_t, std::uint32_t(cpp_dec_float_elem_number)>,
```

Comment:
This looks just plain wrong to me - the 2nd template argument should be of type `cpp_dec_float_elem_number`, unless I'm mistaken you've made it type "function returning std::uint32_t".  Maybe I'm missing something because there's no way this should have got through CI?

Reply:
Ugh, my bad, I realise it's a cast now - Doh!

Could we make this a static_cast please for numpties like me ;)

Reply:
> make this a static_cast please

Good idea. Matt in upcoming changes could you please do that?